### PR TITLE
clean old CODEOWNERS rules

### PR DESCRIPTION
**Pull request by misteriaud**
### What does this PR do?
This PR removes outdated rules in `.github/CODEOWNERS`.
This is the preliminary step before activating a linter to automatically check the correctness of the CODEOWNERS file (#29695).

### CODE REVIEW

**Review by CelianR — APPROVED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -42,7 +42,6 @@
 /repository.datadog.yml                 @DataDog/agent-devx-infra
 /generate_tools.go                      @DataDog/agent-devx-infra
 /service.datadog.yaml                   @DataDog/agent-delivery
-/static-analysis.datadog.yaml           @DataDog/software-integrity-and-trust @DataDog/agent-devx-infra
```
Comment (pgimalac) [inline .github/CODEOWNERS:45]: I believe this should be `/static-analysis.datadog.yml`

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -98,19 +94,15 @@
 /.gitlab/common/test_infra_version.yml               @DataDog/agent-devx-loops @DataDog/agent-devx-infra
 
 /.gitlab/e2e/e2e.yml                                 @DataDog/container-integrations @DataDog/agent-devx-loops
-/.gitlab/e2e/install_packages                        @DataDog/agent-delivery
```
Comment (pgimalac) [inline .github/CODEOWNERS:101]: I think this one should be `.gitlab/e2e_install_packages`

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -98,19 +94,15 @@
...
 /.gitlab/container_build/fakeintake.yml              @DataDog/agent-e2e-testing @DataDog/agent-devx-loops
 /.gitlab/binary_build/fakeintake.yml                 @DataDog/agent-e2e-testing @DataDog/agent-devx-loops
 
 /.gitlab/functional_test/security_agent.yml          @DataDog/agent-security @DataDog/agent-devx-infra
 /.gitlab/functional_test/serverless.yml              @DataDog/serverless @Datadog/serverless-aws @DataDog/agent-devx-infra
-/.gitlab/functional_test_cleanup.yml                 @DataDog/agent-security @DataDog/windows-kernel-integrations @DataDog/agent-devx-infra
```
Comment (pgimalac) [inline .github/CODEOWNERS:107]: There is a `.gitlab/functional_test_cleanup/functional_test_cleanup.yml` so you can probably replace with `.gitlab/functional_test_cleanup`

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -133,7 +125,6 @@
 
 /.gitlab/container_scan/container_scan.yml           @DataDog/container-integrations @DataDog/agent-delivery
 
-/.gitlab/internal_image_deploy.yml                   @DataDog/container-integrations @DataDog/agent-delivery
```
Comment (pgimalac) [inline .github/CODEOWNERS:136]: Same as above there is `.gitlab/powershell_script_deploy/powershell_script_deploy.yml` so you can probably just replace with `.gitlab/powershell_script_deploy`

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -222,7 +206,6 @@
 /Dockerfiles/agent/entrypoint.d.windows/ @DataDog/container-integrations @DataDog/windows-agent
 /Dockerfiles/agent/entrypoint.ps1        @DataDog/container-integrations @DataDog/windows-agent
 /Dockerfiles/agent/windows/              @DataDog/container-integrations @DataDog/windows-agent
-/Dockerfiles/agent-ot/Dockerfile.byoc   @DataDog/opentelemetry
```
Comment (pgimalac) [inline .github/CODEOWNERS:225]: Probably just `/Dockerfiles/agent-ot`

**Review by pgimalac — APPROVED** [review]

File: .github/CODEOWNERS [hunk 6]
```diff
@@ -98,19 +94,15 @@
...
 
 /.gitlab/functional_test/security_agent.yml          @DataDog/agent-security @DataDog/agent-devx-infra
 /.gitlab/functional_test/serverless.yml              @DataDog/serverless @Datadog/serverless-aws @DataDog/agent-devx-infra
-/.gitlab/functional_test_cleanup.yml                 @DataDog/agent-security @DataDog/windows-kernel-integrations @DataDog/agent-devx-infra
-/.gitlab/functional_test/system_probe_windows.yml    @DataDog/agent-devx-infra @DataDog/windows-kernel-integrations
 /.gitlab/functional_test/common.yml                  @DataDog/agent-devx-infra @DataDog/windows-kernel-integrations
 /.gitlab/functional_test/oracle.yml                  @DataDog/agent-devx-infra @DataDog/database-monitoring
 
 /.gitlab/powershell_script_deploy/powershell_script_deploy.yml  @DataDog/agent-delivery @DataDog/windows-agent
-/.gitlab/powershell_script_deploy/powershell_script_signing.yml @DataDog/agent-delivery @DataDog/windows-agent
```
Comment (pgimalac) [inline .github/CODEOWNERS:113]: Same as above there is `.gitlab/powershell_script_deploy/powershell_script_deploy.yml` so you can probably just replace with `.gitlab/powershell_script_deploy`

**Review by pgimalac — APPROVED** [review]

**Pull request merged**